src/e-build.js

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const program = require('commander');
+
+const evmConfig = require('./evm-config');
+const { depot, sccache, fatal } = require('./util');
+
+function runGNGen(config) {
+  depot.ensure();
+
+  const gn_args = config.gen.args.join(' ').replace(/\"/g, '\\"'); // gn parses this part -- inner quotes must be escaped
+  const exec = `${path.resolve(depot.path, 'gn.py')} gen "out/${
+    config.gen.out
+  }" --args="${gn_args}"`;
+  const opts = { cwd: path.resolve(config.root, 'src') };
+  depot.execSync(config, exec, opts);
+}
+
+function ensureGNGen(config) {
+  const buildfile = path.resolve(evmConfig.outDir(config), 'build.ninja');
+  if (!fs.existsSync(buildfile)) runGNGen(config);
+}
+
+function runNinja(config, target, ninjaArgs) {
+  sccache.ensure(config);
+  depot.ensure(config);
+  ensureGNGen(config);
+
+  const exec = os.platform === 'win32' ? 'ninja.exe' : 'ninja';
+  const args = [...ninjaArgs, target];
+  const opts = { cwd: evmConfig.outDir(config) };
+  depot.execFileSync(config, exec, args, opts);
+}
+
+program
+  .allowUnknownOption()
+  .arguments('[target] [ninjaArgs...]')
+  .description('Build Electron and other things.')
```

Comment:
"other executables" isn't perfect, since targets include things like "node:headers"

I've changed this to "Build Electron and other targets" as a cue to tie it to the other uses of 'target' in the help message, e.g.`--list-targets`

Changed d7d0ea9a6213e897d375ef618b95a8e6ee7e42f1